Jenkinsfile

```diff
@@ -358,177 +326,155 @@ gcloud container clusters create ${clusterName} \
-                    withGo() {
-                        dir('src/github.com/bitnami/kube-prod-runtime') {
-                            // NB: `kubeprod` also uses az cli credentials and
-                            // $AZURE_SUBSCRIPTION_ID, $AZURE_TENANT_ID.
-                            withCredentials([azureServicePrincipal('jenkins-bkpr-owner-sp')]) {
-                                def resourceGroup = 'jenkins-bkpr-rg'
-                                def clusterName = ("${env.BRANCH_NAME}".take(8) + "-${env.BUILD_NUMBER}-" + UUID.randomUUID().toString().take(5) + "-${platform}").replaceAll(/[^a-zA-Z0-9-]/, '-').replaceAll(/--/, '-').toLowerCase()
-                                def dnsPrefix = "${clusterName}"
-                                def dnsZone = "${dnsPrefix}.${parentZone}"
-                                def adminEmail = "${clusterName}@${parentZone}"
-                                def location = "eastus"
-                                def availableK8sVersions = ""
-                                def kversion_full = ""
-
-                                try {
-                                    runIntegrationTest(platform, "aks --config=${clusterName}-autogen.json --dns-resource-group=${resourceGroup} --dns-zone=${dnsZone} --email=${adminEmail}", "--dns-suffix ${dnsZone}") {
-                                        container('az') {
-                                            sh '''
-az login --service-principal -u $AZURE_CLIENT_ID -p $AZURE_CLIENT_SECRET -t $AZURE_TENANT_ID
-az account set -s $AZURE_SUBSCRIPTION_ID
-'''
-                                            availableK8sVersions = sh(returnStdout: true, script: "az aks get-versions --location ${location} --query \"orchestrators[?contains(orchestratorVersion,'${kversion}')].orchestratorVersion\" -o tsv")
-                                        }
+                // See:
+                //  az aks get-versions -l centralus --query 'sort(orchestrators[?orchestratorType==`Kubernetes`].orchestratorVersion)'
+                def aksKversions = ["1.10", "1.11"]
+                for (kversion in aksKversions) {
+                    def resourceGroup = 'jenkins-bkpr-rg'
+                    def location = "eastus"
+                    def platform = "aks-" + kversion
+
+                    platforms[platform] = {
+                        stage(platform) {
+                            def clusterName = ("${env.BRANCH_NAME}".take(8) + "-${env.BUILD_NUMBER}-" + UUID.randomUUID().toString().take(5) + "-${platform}").replaceAll(/[^a-zA-Z0-9-]/, '-').replaceAll(/--/, '-').toLowerCase()
+                            def dnsPrefix = "${clusterName}"
+                            def adminEmail = "${clusterName}@${parentZone}"
+                            def dnsZone = "${dnsPrefix}.${parentZone}"
+
+                            dir("${env.WORKSPACE}/${clusterName}") {
+                                withEnv(["KUBECONFIG=${env.WORKSPACE}/.kubecfg-${clusterName}"]) {
```

Comment:
should we set `HOME` to this test-specific dir too?

Reply:
The az, gcloud cli's use `HOME` to store the user profile. By using a common `HOME` we are able to perform a login to all services just once.